app/scripts/controllers/metamask-notifications/metamask-notifications.ts

```diff
@@ -281,30 +293,49 @@ export class MetamaskNotificationsController extends BaseController<
     },
   };
 
-  #prevAccountsSet = new Set<string>();
-
   #accounts = {
     /**
      * Used to get list of addresses from keyring (wallet addresses)
      *
      * @returns addresses removed, added, and latest list of addresses
      */
     listAccounts: async () => {
+      // Get previous and current account sets
       const nonChecksumAccounts = await this.messagingSystem.call(
         'KeyringController:getAccounts',
       );
       const accounts = nonChecksumAccounts.map((a) => toChecksumHexAddress(a));
       const currentAccountsSet = new Set(accounts);
+      const prevAccountsSet = new Set(this.state.subscriptionAccountsSeen);
+
+      // Invalid value you cannot have zero accounts
+      // Only occurs when the Accounts controller is initializing.
+      if (accounts.length === 0) {
+        return {
+          accountsAdded: [],
+          accountsRemoved: [],
+          accounts: [],
+        };
+      }
 
-      const accountsAdded = accounts.filter(
-        (a) => !this.#prevAccountsSet.has(a),
-      );
-
-      const accountsRemoved = [...this.#prevAccountsSet.values()].filter(
+      // Calculate added and removed addresses
+      const accountsAdded = accounts.filter((a) => !prevAccountsSet.has(a));
+      const accountsRemoved = [...prevAccountsSet.values()].filter(
         (a) => !currentAccountsSet.has(a),
       );
 
-      this.#prevAccountsSet = new Set(accounts);
+      console.log({
```

Comment:
console.log should be removed

Reply:
Also good spot, this was a late PR so can fix CI and ask for re-review